### PR TITLE
Add cartesian_to_spherical().

### PR DESCRIPTION
**Pull request by MLopez-Ibanez**

### CODE REVIEW

File: tests/tests_core/test_core_util.py [hunk 1]
```diff
@@ -32,19 +37,30 @@ def test_rotation_matrix_z():
...
+    sph_car(
+        np.array([[0.5, np.pi / 4, -np.pi / 4], [0.5, -np.pi / 4, np.pi / 4]]),
+        np.array([[0.25, -0.25, 0.35355339], [-0.25, -0.25, 0.35355339]]),
+    )
 
-    result = spherical_to_cartesian(
-        np.array([[0.5, np.pi / 4, -np.pi / 4], [0.5, -np.pi / 4, np.pi / 4]])
+    sph_car(
+        np.array([2.60564963, 1.75305207, 4.4458828]),
+        np.array([-0.674864797187, -2.472029259161, -0.472269863940]),
```
[the remark this answers is not in the document]
Comment (MLopez-Ibanez) [reply tests/tests_core/test_core_util.py:62]: I have no experience with pytest so I cannot give an opinion. But these particular values were chosen because they trigger corner cases due to the use of arctan2 and arccos. There are probably nicer ways to test the corner cases but I didn't have much time to think about them.

[the remark this answers is not in the document]
Comment (MLopez-Ibanez) [reply tests/tests_core/test_core_util.py:62]: In fact, if there was a way to sample the full range of values in terms of positive/negative directions, that would be even better.

[the remark this answers is not in the document]
Comment (astrojuanlu) [reply tests/tests_core/test_core_util.py:62]: Right! I'll have a look at this and try a version with Hypothesis + pytest then
